### PR TITLE
[protocol][feat] add RPC query endpoint for withdrawal gating information

### PR DESCRIPTION
**Pull request by lucas-dydx**
### Changelist
Add RPC query endpoint for withdrawal gating information.

### Test Plan
Added unit tests, and will use when testing withdrawal gating E2E in `dev` / `staging` env.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	ctx := sdktypes.UnwrapSDKContext(c)
+
+	downtimeInfo := k.blocktimeKeeper.GetDowntimeInfoFor(
+		ctx,
+		types.WITHDRAWAL_AND_TRANSFERS_BLOCKED_AFTER_CHAIN_OUTAGE_DURATION,
+	)
+	chainOutageSeenAtBlock, chainOutageExists := downtimeInfo.BlockInfo.Height, downtimeInfo.BlockInfo.Height > 0
+	negativeTncSubaccountSeenAtBlock, negativeTncSubaccountSeenAtBlockExists := k.GetNegativeTncSubaccountSeenAtBlock(ctx)
+
+	// Withdrawals and transfers are blocked at the
```
Comment (jonfung-dydx) [inline protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go:29]: finish your thought 👑

File: protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	withdrawalsAndTransfersBlockedUntilBlock := uint32(0)
```
Comment (roy-dydx) [inline protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go:30]: Blocked until block 0 means no blocking right?

**Review by roy-dydx — APPROVED** [review]
blocktime usage looks good to me.

File: protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go (hunk 2 above)
Comment (lucas-dydx) [reply protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go:30]: Yep, essentially means withdrawals and transfers are unblocked at block 0 (not blocked)

File: protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go (hunk 1 above)
Comment (lucas-dydx) [reply protocol/x/subaccounts/keeper/grpc_query_withdrawal_and_transfers_blocked_info.go:29]: 😭

**Review by jonfung-dydx — APPROVED** [review]

**Pull request merged**